File: mobile-wallet-master/mifospay/src/main/java/org/mifos/mobilewallet/mifospay/bank/presenter/DebitCardPresenter.java
package org.mifos.mobilewallet.mifospay.bank.presenter;

import org.mifos.mobilewallet.mifospay.bank.BankContract;
import org.mifos.mobilewallet.mifospay.base.BaseView;

import javax.inject.Inject;

/**
 * Created by ankur on 13/July/2018
 */

public class DebitCardPresenter implements BankContract.DebitCardPresenter {

    BankContract.DebitCardView mDebitCardView;

    @Inject
    public DebitCardPresenter() {

    }

    @Override
    public void attachView(BaseView baseView) {
        mDebitCardView = (BankContract.DebitCardView) baseView;
        mDebitCardView.setPresenter(this);
    }

    @Override
    public void verifyDebitCard(String s, String s1, String s2) {
        String otp = "0000";
        mDebitCardView.verifyDebitCardSuccess(otp);
//        String message = "error";
//        mDebitCardView.verifyDebitCardError(message);
    }
}

File: mifospay/src/main/java/org/mifospay/bank/presenter/DebitCardPresenter.kt
package org.mifospay.bank.presenter

import org.mifospay.MifosPayApp
import org.mifospay.R
import org.mifospay.bank.BankContract
import org.mifospay.bank.BankContract.DebitCardView
import org.mifospay.base.BaseView
import javax.inject.Inject

/**
 * Created by ankur on 13/July/2018
 */
class DebitCardPresenter @Inject constructor() : BankContract.DebitCardPresenter {
    var mDebitCardView: DebitCardView? = null
    override fun attachView(baseView: BaseView<*>?) {
        mDebitCardView = baseView as DebitCardView?
        mDebitCardView!!.setPresenter(this)
    }

    override fun verifyDebitCard(s: String?, s1: String?, s2: String?) {
        val otp = "0000"
        if (s!!.length < 12 || s.length > 19) {
            mDebitCardView!!.verifyDebitCardError(
                MifosPayApp.context
                    ?.getString(R.string.debit_card_error_message)
            )
        } else {
            mDebitCardView!!.verifyDebitCardSuccess(otp)
        }
    }
}